cfme/utils/config.py

```diff
@@ -0,0 +1,140 @@
+    def __init__(self, env='cfme_data'):
+        self.settings = LazySettings(
+            ENV_FOR_DYNACONF=env,
+            INCLUDES_FOR_DYNACONF='{}/*.yaml'.format(path.conf_path.strpath),
+            ROOT_PATH_FOR_DYNACONF=path.conf_path.strpath
+        )
+
+    def __getitem__(self, key):
+        return self.__getattr__(key)
+
+    def __getattr__(self, key):
+        self.config = self.settings[key.upper()]
+        return self.config
+
+    def __setitem__(self, key, value):
+        """This function allows you to mimic some behavior like runtime from yaycl
+            You can do following:
+                from cfme.utils import conf
+                conf ['env']['new_key'] = 'new_val'
+                conf ['env']['new_key']
+                # The new_key and its value is cached through the session
+        """
+        self.settings[key] = value
+
+    def __contains__(self, key):
+        return key in self.settings.as_dict() or key.upper() in self.settings.as_dict()
+
+    def get(self, key, default=None):
+        try:
+            return self.__getattr__(key)
+        except KeyError:
+            return default
+
+
+class Configuration(object):
+    """
+    Holds ConfigData objects in dictionary `settings`.
+    """
+    def __init__(self):
+        self.settings = None
+        self.yaycl_config = None
+
+    def configure_creds(self, config_dir, crypt_key_file=None):
+        """
+        do the defered initial loading of the configuration
+
+        :param config_dir: path to the folder with configuration files
+        :param crypt_key_file: optional name of a file holding the key for encrypted
+            configuration files
+
+        :raises: AssertionError if called more than once
+
+        if the `utils.conf` api is removed, the loading can be transformed to eager loading
+        """
+        assert self.yaycl_config is None
+        if crypt_key_file and os.path.exists(crypt_key_file):
+            self.yaycl_config = yaycl.Config(
+                config_dir=config_dir,
+                crypt_key_file=crypt_key_file)
+        else:
+            self.yaycl_config = yaycl.Config(config_dir=config_dir)
+
+    def configure(self):
+        """
+        Loads all the congfigs in settings dictionary where Key is filename(without extension)
+        and value is ConfigData object relevant to that file.
+
+        Example self.settings would look like:
+        {'cfme_data': <cfme.utils.config_copy.ConfigData at 0x7f2a943b14e0>,
+         'env': <cfme.utils.config_copy.ConfigData at 0x7f2a943b1be0>,
+         'gpg': <cfme.utils.config_copy.ConfigData at 0x7f2a943b1cc0>,
+         ...
+         ...
+         'polarion_tools': <cfme.utils.config_copy.ConfigData at 0x7f2a943ad438>,
+         'perf_tests': <cfme.utils.config_copy.ConfigData at 0x7f2a943ad2e8>}
+
+        """
+        if self.settings is None:
+            self.settings = {
+                file.basename[:-5]: ConfigData(env=file.basename[:-5])
```

Comment:
Why the `-5`? Maybe add a comment here?

Reply:
.yml is also a valid file type, this is a bad practice.

Since `file` here is a Py.path.local instance, you want to use `file.purebasename`.

Reply:
```
s = {f.purebasename: 'a value' for f in conf_path.listdir() if '.local.yaml' not in f.basename and f.ext in ['.yaml', '.yml']}
```

Purebasename alone gets tripped up by the .local.yaml files, (as you pointed out in chat! :) ) but you should be fine using it like above since you're filtering the files anyway within the comprehension.

Reply:
fixed this